Examples/Algorithms/TrackFitting/src/TrackFittingAlgorithm.cpp

```diff
@@ -159,9 +167,11 @@ ActsExamples::ProcessCode ActsExamples::TrackFittingAlgorithm::execute(
     }
   }
 
-  std::stringstream ss;
-  trackStateContainer->statistics().toStream(ss);
-  ACTS_DEBUG(ss.str());
+  if (logger().doPrint(Acts::Logging::DEBUG)) {
+    std::stringstream ss;
+    trackStateContainer->statistics().toStream(ss);
+    ACTS_DEBUG(ss.str());
+  }
```

Comment:
```suggestion
  ACTS_DEBUG([](){
    std::stringstream ss;
    trackStateContainer->statistics().toStream(ss);
    return ss.str();
  }());
```

Also, maybe this could be `VERBOSE`?

Reply:
I think `DEBUG` is preferable because `VERBOSE` will contain all the propagation verbose logs

to me the `if` looks easier to read but the lambda looks interesting. I did't think of that